app/services/search/postgres/comment.rb

```diff
@@ -0,0 +1,89 @@
+module Search
+  module Postgres
+    class Comment
+      ATTRIBUTES = [
+        "articles.published",
+        "comments.body_markdown",
+        "comments.commentable_id AS commentable_id",
+        "comments.commentable_type AS commentable_type",
+        "comments.created_at",
+        "comments.id AS id",
+        "comments.public_reactions_count",
+        "comments.score",
+        "comments.user_id AS comment_user_id",
+        "podcast_episodes.podcast_id AS podcast_id",
+        "podcasts.published",
+        "users.id AS user_id",
+        "users.name",
+        "users.profile_image",
+        "users.username",
+      ].freeze
+      private_constant :ATTRIBUTES
+
+      DEFAULT_PER_PAGE = 60
+      private_constant :DEFAULT_PER_PAGE
+
+      # Because commentable represents a polymorphic relationship, ActiveRecord
+      # can't eager load the associations so we have to make all the joins
+      # manually.
+      #
+      # NOTE: if Comment::COMMENTABLE_TYPES is updated, this filter will also
+      # need to be updated
+      FORCED_EAGER_LOAD_QUERY = <<-SQL.freeze
+        LEFT JOIN users
+          ON comments.user_id = users.id
+        LEFT JOIN articles
+          ON comments.commentable_id = articles.id
+          AND comments.commentable_type = 'Article'
+        LEFT JOIN podcast_episodes
+          ON comments.commentable_id = podcast_episodes.id
+          AND comments.commentable_type = 'PodcastEpisode'
+        LEFT JOIN podcasts
+          ON podcast_episodes.podcast_id = podcasts.id
+      SQL
+      private_constant :FORCED_EAGER_LOAD_QUERY
```

Comment:
As mentioned [here](https://github.com/forem/forem/pull/13401/files#r613424752), this is due to polymorphism. Essentially what we're trying to do is eager load the necessary associations. But, because the relationship is polymorphic, the "regular" approaches of `includes` and `preload` don't work here. ActiveRecord doesn't know how to handle the references. I did some research and it seems the most common workaround is to make the joins yourself. While this is a little more complicated and requires us to maintain this query anytime we change what's a valid `commentable` object, it does give us more flexibility in these queries.

Unfortunately, `PodcastEpisode` needs some fields from `Podcast` like `published`. So it sort of looks like:
```
Comment --> PodcastEpisode --> Podcast
Comment --> Article
```

If/when we ever add a new object that is `commentable` that we want to search against, we'll have to manually update the query and filter. I tried adding code comments to make this clear.